internal/hscan/structmap.go

```diff
@@ -97,7 +100,10 @@ func (s StructValue) Scan(key string, value string) error {
 	}
 
 	if isPtr && v.Type().NumMethod() > 0 && v.CanInterface() {
-		if scan, ok := v.Interface().(Scanner); ok {
+		switch scan := v.Interface().(type) {
+		case encoding.TextUnmarshaler:
+			return scan.UnmarshalText(util.StringToBytes(value))
+		case Scanner:
```

Comment:
Perhaps swap Scanner and TextUnmarshaler? Scanner looks more specific.

Reply:
ah